client_plugin/drivers/vfile/kvstore/etcdops/etcdops.go

```diff
@@ -1043,6 +1271,69 @@ func (e *EtcdKVS) WriteMetaData(entries []kvstore.KvPair) error {
 	return nil
 }
 
+// UpdateMetaData - Read/Write/Delete metadata according to given key-value pairs
+func (e *EtcdKVS) UpdateMetaData(entries []kvstore.KvPair) ([]kvstore.KvPair, error) {
+	var ops []etcdClient.Op
+	//var result_entries []kvstore.KvPair
+	log.WithFields(
+		log.Fields{"KvPair": entries},
+	).Debug("UpdateMetaData")
+
+	// Create a client to talk to etcd
+	client := e.createEtcdClient()
+	if client == nil {
+		return nil, errors.New(etcdClientCreateError)
+	}
+	defer client.Close()
+
+	// Lets build the request which will be executed
+	// in a single transaction
+	// ops contain multiple operations
+	for _, elem := range entries {
+		if elem.OpType == kvstore.OpPut {
+			ops = append(ops, etcdClient.OpPut(elem.Key, elem.Value))
+		} else if elem.OpType == kvstore.OpGet {
+			ops = append(ops, etcdClient.OpGet(elem.Key))
+		} else if elem.OpType == kvstore.OpDelete {
+			ops = append(ops, etcdClient.OpDelete(elem.Key))
+		} else {
+			msg := fmt.Sprintf("Unknown OpType for UpdateMetaData: %s", elem.OpType)
+			log.Errorf(msg)
+			return nil, errors.New(msg)
+		}
+	}
+
+	// Post all requested operations in one transaction
+	ctx, cancel := context.WithTimeout(context.Background(), etcdRequestTimeout)
+	resp, err := client.Txn(ctx).Then(ops...).Commit()
+	cancel()
+	if err != nil {
+		msg := fmt.Sprintf("Transactional metadata update failed: %v.", err)
+		if err == context.DeadlineExceeded {
+			msg += fmt.Sprintf(swarmUnhealthyErrorMsg)
```

Comment:
Why here we add error for "swarmUnhealthy"?

Reply:
Should be etcdUnhealthy. Will update.